sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -547,10 +548,15 @@ class HiveContext(sc: SparkContext) extends SQLContext(sc) with Logging {
     )
   }
 
+  @transient
+  private lazy val matcher = Pattern.compile(
```

Comment:
`matcher` is kind of a generic name; can you come up with something more descriptive, like `functionOrMacroDDLPattern`?